### PR TITLE
[New Package] Add Restic v0.12.1

### PR DESCRIPTION
**Pull request by alecbcs**
Restic is a great backup program written in Go that handles secure end-to-end encrypted backups that can be uploaded to multiple cloud providers automatically. If you're interested checkout [restic.net](https://restic.net) for more information.

### CODE REVIEW

File: var/spack/repos/builtin/packages/restic/package.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    version('0.12.1', sha256='a9c88d5288ce04a6cc78afcda7590d3124966dab3daa9908de9b3e492e2925fb')
+
+    depends_on("go", type='build')
+
+    def setup_build_environment(self, env):
+        # Point GOPATH at the top of the staging dir for the build step.
+        env.prepend_path('GOPATH', self.stage.path)
+
+    def install(self, spec, prefix):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/restic/package.py]: The following should fix the style error:

```suggestion
    def install(self, spec, prefix):
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the sha256.  Needs the stye error resolved.

**Comment by tldahlgren** [comment]
You probably also want to rebase to develop to avoid an issue with the pipeline that was fixed earlier today.

**Comment by alecbcs** [comment]
@tldahlgren thanks for the catch! So funny a space was hiding out at the end of a line! 

Happy to rebase in just a minute when I get back to a desktop!

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**